src/pymor/analyticalproblems/expressions.py

```diff
@@ -304,13 +373,18 @@ def __init__(self, base, index):
             raise ValueError(f'Wrong number of indices (given: {index} for expression "{base}" of shape {base.shape})')
         if not all(0 <= i < s for i, s in zip(index, base.shape)):
             raise ValueError(f'Invalid index (given: {index} for expression "{base}" of shape {base.shape})')
-        self.base, self.index = base, index
+        self.base, self.index = _convert_to_expression(base), index
         self.shape = base.shape[len(index):]
 
     def numpy_expr(self):
         index = ['...'] + [repr(i) for i in self.index]
         return f'{self.base.numpy_expr()}[{",".join(index)}]'
 
+    def fenics_expr(self, params):
+        if len(self.base.shape) != 1:
+            raise NotImplementedError
+        return np.array(self.base.fenics_expr(params)[self.index[0]])
```

Comment:
If I'm not overlooking something, since `fenics_expr` now always returns an array, you should be able to write

```python
return self.base.fenics_expr(params)[self.index]
```